sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +633,24 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(javaOptions: Seq[String]): Option[VirtualJSFile] = {
```

Comment:
This should probably be in its own `Task`, so that it can be reused/overridden by users. It seems a bit too monolithic right now.